pilot/pkg/bootstrap/server.go

```diff
@@ -797,7 +802,22 @@ func (s *Server) initRegistryEventHandlers() {
 	s.ServiceController().AppendServiceHandler(serviceHandler)
 
 	if s.configController != nil {
-		configHandler := func(_, curr config.Config, event model.Event) {
+		configHandler := func(old config.Config, curr config.Config, event model.Event) {
+			if old.Generation == curr.Generation && curr.Generation != 0 {
+				// Kubernetes will start generation at 1, but some internally generated configurations
+				// may not set resource version at all and we still want updates from these
+				if curr.GroupVersionKind == gvk.WorkloadEntry {
+					oldCond := modelstatus.GetConditionFromSpec(old, modelstatus.ConditionHealthy)
+					newCond := modelstatus.GetConditionFromSpec(curr, modelstatus.ConditionHealthy)
+					if oldCond == newCond {
+						return
+					}
+				} else if onlyStatusUpdated(old, curr) {
+					log.Debugf("skipping push for %v %v\n", old, curr)
```

Comment:
Even at debug log, logging the full config object will be massive. Can we do just the name? and indicate why we skiped it?